### PR TITLE
Add Tasks SQL Powered to Community Plugin List

### PR DESCRIPTION
**Pull request by sytone**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/sytone/obsidian-tasks-x

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [x]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by WhiteNoiseDev** [comment]
If the goal of this fork is to be merged with the obsidian tasks at some point, wouldn't be better to just let this fork live in BRAT?

**Comment by sytone** [comment]
@WhiteNoiseDev I have considered that but there is no timeline with the final merge and wanted to make sure people could find it along side the current Tasks to make sure it was clear that this is a vNext as it is backwards compatible and so the current Tasks project does not get questions based on the new functionality.

**Comment by sytone** [comment]
Also validated and running on iOS and Android.

**Comment by sytone** [comment]
following up on this to get it approved. The merge is not going to happen soon and want it to be easy for others to use it.

**Comment by sytone** [comment]
Fixed conflict ready for merge.

**Comment by claremacrae** [comment]
For info: Just to say that I am currently planning to start merging features from this plugin to the main Tasks repo in the next 2 to 4 weeks.

I'll be re-structuring and reformatting the tasks code to match the layout changes in tasks-x, so that the diffing of files, and the merging in of features, will be very manageable.

The only reason I haven't yet started merging features in is that I have first been working through the outstanding PRs in Tasks, to merge those. Reviewing the PRs before doing the moving of source files and changing of line-length has made the PRs easier to deal with.

The merge will be incremental, so for example I might start be migrating in the ability to have custom Task states, as this is a very highly requested feature, and I know that sytone has been working on this for some time.

**Comment by claremacrae** [comment]
But if this PR is to be merged, I would really appreciate it if the following incompatibility with the behaviour of the Tasks plugin could be fixed first:

https://github.com/sytone/obsidian-tasks-x/issues/10

Otherwise, users would not be able to switch between the two plugins with consistent behaviour, and I slightly fear for the user support burden on Tasks...

**Comment by WhiteNoiseDev** [comment]
From the users POV, it's better to have 1 plugin with all the features. That being said I am not sure this is possible. 
At this point. I don't think of this as a vNext, I consider this a contentious fork.

I encourage @claremacrae and @sytone to have a private chat on IF and HOW you want to collaborate. If reach an agreement, great. If you don't, that's fine too, we'll merge the fork and will let the users digest the differences that will grow over time.

**Comment by sytone** [comment]
> But if this PR is to be merged, I would really appreciate it if the following incompatibility with the behaviour of the Tasks plugin could be fixed first:
> 
> [sytone/obsidian-tasks-x#10](https://github.com/sytone/obsidian-tasks-x/issues/10)
> 
> Otherwise, users would not be able to switch between the two plugins with consistent behaviour, and I slightly fear for the user support burden on Tasks... (and more importantly, users not getting the results they expected...)

I have a change for that pending. It is a option to have the default flow [ ] -> [x] or [ ] -> [/] -> [x]. I have been carefult in the configuration and notes to be explicit where to file issues and been keeping a eye on tasks changes and issues.

**Comment by sytone** [comment]
> 

I am keeping the new implementation in sync with tasks capability and features as well as trying to feedback when I have time. It is approaching things differntly and at a higher speed of change which is why it came about in the first place.

**Comment by claremacrae** [comment]
Quick update:

- Just to note that in the next Tasks release I intend to announce a date for my restructuring and reformatting code in the Tasks plugin, which will help with later 2-way merging between the two plugins
- After that, I will start offering individual PRs to merge a few of the recent new tasks features in to tasks-x, to hopefully save sytone some time in doing that, and keep it safe...
- sytone has kindly already made two PRs to start merge from tasks-x to tasks (one is awaiting feedback on code review comments)

**Comment by claremacrae** [comment]
> I encourage @claremacrae and @sytone to have a private chat on IF and HOW you want to collaborate. If reach an agreement, great. If you don't, that's fine too, we'll merge the fork and will let the users digest the differences that will grow over time.

We did actually speak some time ago.

File: community-plugins.json [hunk 1]
```diff
@@ -4558,5 +4558,13 @@
         "description": "Allows you to hide files and folders from the Obsidian file explorer if you don't want to see them anymore.",
         "repo": "Oliver-Akins/file-hider",
         "branch": "main"
+    },
+    {
+        "id": "obsidian-tasks-x-plugin",
+        "name": "Tasks SQL Powered",
+        "author": "Jon Bullen (original by Martin Schenck)",
```
Comment (claremacrae) [inline community-plugins.json:4565]: I would appreciate that saying `(original by Martin Schenck and Clare Macrae)` please.

**Comment by claremacrae** [comment]
@sytone wrote:

> I am keeping the new implementation in sync with tasks capability and features as well as trying to feedback when I have time. It is approaching things differntly and at a higher speed of change which is why it came about in the first place.

I've done a bunch or rearranging and reformatting of the Tasks code today, and diff'ing the two code bases, and see now that there are still significant differences - both plugins have major improvements that the other one does not.

Even though we are going to try and keep them in sync, it's clear that a lot of work will be required to do that - and so  if @sytone would still like this to be added to the community plugin list and it's still going to be maintained, I would now very much support that request. It would make it easier for him to get feedback on his work.

I think Tasks users will get a better experience, indirectly, from what I am learning from sytone's work, and I would like to support that.

**So - in a bit of a U-turn - I would support this being merged, if sytone still wishes that to happen...**

**Comment by claremacrae** [comment]
I just noticed this old comment from me...

> But if this PR is to be merged, I would really appreciate it if the following incompatibility with the behaviour of the Tasks plugin could be fixed first:
> 
> [sytone/obsidian-tasks-x#10](https://github.com/sytone/obsidian-tasks-x/issues/10)
> 
> Otherwise, users would not be able to switch between the two plugins with consistent behaviour, and I slightly fear for the user support burden on Tasks... (and more importantly, users not getting the results they expected...)

I think it's still relevant...

**Comment by liamcain** [comment]
Hi @sytone, I'm sorry this pull request has been in review limbo for so long. I just wanted to check in to make sure you still wanted to see this plugin included in the plugin gallery.

I think we're fine with including this as is, as long as you update the author field to include Clare's name. Thanks!

**Comment by joethei** [comment]
Closing PR due to inactivity.

Feel free to reopen this PR if you still intend on submitting this plugin.

**Pull request closed without merging**